### PR TITLE
Roll around the correct aimpoint, redux

### PR DESCRIPTION
**Pull request by taldcroft**
Sparkles is sparking no joy right now...

79aff0615ded0 only did half the job.  This finishes it up.  For validation work this is based off of pickle-roll-options branch, see ad8c820.

### CODE REVIEW

File: sparkles/roll_optimize.py [hunk 1]
```diff
@@ -309,7 +309,7 @@ def improve_metric(n_stars, P2, n_stars_new, P2_new):
         for roll_interval in roll_intervals:
             roll = roll_interval['roll']
             q_targ_roll = Quat([q_targ.ra, q_targ.dec, roll])
-            q_att_roll = calc_aca_from_targ(q_targ_roll, 0, 0)
+            q_att_roll = self._calc_aca_from_targ(q_targ_roll, 0, 0)
```
Comment (jeanconn) [inline sparkles/roll_optimize.py]: Looks great.  Though I'm assuming if this wasn't noticed that this could use a quick test.  I could help with that.

Comment (taldcroft) [reply sparkles/roll_optimize.py]: Sure, thanks!

Comment (jeanconn) [reply sparkles/roll_optimize.py]: I think this would also be more readable if those were q_att_rolled and q_targ_rolled, but that might just be me.

**Comment by jeanconn** [comment]
Also, I haven't done the math, but does the fix in ad8c820 mean that the fix in 91267be is still required?

File: sparkles/roll_optimize.py (hunk 1 above)
Comment (taldcroft) [reply sparkles/roll_optimize.py]: Done.

**Comment by taldcroft** [comment]
> Also, I haven't done the math, but does the fix in [ad8c820](https://github.com/sot/sparkles/commit/ad8c8200dafec1a9936420b9d72078fc7fe79363) mean that the fix in [91267be](https://github.com/sot/sparkles/commit/91267bee86609f321e785aa93d7887933b8946a0) is still required?

The 91267be fix is still necessary.  In the case of a 0.0 allowed roll deviation (which was the case for the original bug), it will always be true that the ORviewer supplied `roll` is not within the `roll_min` to `roll_max` range, because that range had zero width (before the fix), and ORviewer is using a different algorithm for allowed roll range.

**Comment by taldcroft** [comment]
@jeanconn - can you take care of getting the open/approved PRs merged?  I don't care if you do the individual ones or just this PR which is the glop of them together.  I'm good with all.

**Comment by taldcroft** [comment]
And I'm going to start work on the target roll stuff based on the roll-opt-targ-bug branch, so it'll be nice to get this all into master.

**Pull request merged**